Framework/API/src/Algorithm.cpp

```diff
@@ -700,6 +706,29 @@ void Algorithm::executeAsChildAlg() {
   }
 }
 
+void Algorithm::exec(MPI::ExecutionMode executionMode) {
+  switch (executionMode) {
+  case MPI::ExecutionMode::Serial:
+  case MPI::ExecutionMode::Identical:
+    return exec();
+  case MPI::ExecutionMode::Distributed:
+    return execDistributed();
+  case MPI::ExecutionMode::MasterOnly:
+    return execMasterOnly();
+  default:
+    throw(std::runtime_error("Algorithm " + name() +
+                             " does not support execution mode " +
+                             MPI::toString(executionMode)));
+  }
+}
+
+void Algorithm::execMasterOnly() {
+  if (MPI::isRoot())
+    return exec();
+  else
+    return execNonMaster();
```

Comment:
Not entirely sure what this is going to do. This has to do with your previous comment regarding setting properties (We'll talk on the phone about this)